include/mata/nfa.hh

```diff
@@ -477,8 +479,7 @@ public:
      */
     StateSet get_useful_states_old() const;
 
-    //I just want to return something as constant reference to the thing, without copying anything, how?
-    const Util::NumberPredicate<State> get_useful_states() const;
+    std::vector<char> get_useful_states() const;
```

Comment:
No really lets not do useless expensive conversions. The programmer is mainly us. But we can rename it, it would make sense to have a global name for a vector of boolean flags.

Reply:
If we do not want to support that, then I am fine with the current state. But I would rename the type, as you say.